### PR TITLE
Song selection screen 

### PR DESCRIPTION
**Pull request by CallMeAlasca**
Linked Task: #RASPIO-31

### CODE REVIEW

File: app/app/src/main/java/me/bickositieff/raspio/ui/playlist/SongAdapter.kt [hunk 1]
```diff
@@ -6,7 +6,7 @@ import androidx.recyclerview.widget.RecyclerView
 import me.bickositieff.raspio.databinding.SongItemBinding
 import me.bickositieff.raspio.ui.models.Song
 
-class SongAdapter(
+open class SongAdapter(
```
Comment (profiluefter) [inline app/app/src/main/java/me/bickositieff/raspio/ui/playlist/SongAdapter.kt]: I don't think this is still needed, is it?
```suggestion
class SongAdapter(
```

File: app/app/src/main/java/me/bickositieff/raspio/ui/songSelect/SongSelectFragment.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.RecyclerView
+import me.bickositieff.raspio.R
+
+class SongSelectFragment : Fragment() {
+
+
+    private val viewModel: SongSelectViewModel by viewModels()
+    private lateinit var mAdapter: SongSelectAdapter
```
Comment (profiluefter) [inline app/app/src/main/java/me/bickositieff/raspio/ui/songSelect/SongSelectFragment.kt:16]: There's no reason for this to be an instance variable. Content is updated via LiveData

File: app/app/src/main/java/me/bickositieff/raspio/ui/songSelect/SongSelectViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import me.bickositieff.raspio.generated.api.PlaylistApi
+import me.bickositieff.raspio.generated.models.POSTPlaylistBody
+import me.bickositieff.raspio.ui.models.Song
+
+class SongSelectViewModel : ViewModel() {
+    val availableSongs = liveData {
+        emit(emptyList())
+        while (true) {
+            emit(LibraryApi.getSongs().body()!!.map { Song(it.path!!, it.title!!, it.artist!!, it.duration!!, null) })
+            kotlinx.coroutines.delay(1000)
```
Comment (profiluefter) [inline app/app/src/main/java/me/bickositieff/raspio/ui/songSelect/SongSelectViewModel.kt]: Add an import
```suggestion
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import me.bickositieff.raspio.generated.api.LibraryApi
import me.bickositieff.raspio.generated.api.PlaylistApi
import me.bickositieff.raspio.generated.models.POSTPlaylistBody
import me.bickositieff.raspio.ui.models.Song

class SongSelectViewModel : ViewModel() {
    val availableSongs = liveData {
        emit(emptyList())
        while (true) {
            emit(LibraryApi.getSongs().body()!!.map { Song(it.path!!, it.title!!, it.artist!!, it.duration!!, null) })
            delay(1000)
```

File: app/app/src/main/res/menu/popup_select_song.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu
```
Comment (profiluefter) [inline app/app/src/main/res/menu/popup_select_song.xml:2]: very interesting formatting in this file

File: app/app/src/main/res/navigation/mobile_navigation.xml [hunk 5]
```diff
@@ -9,7 +9,12 @@
             android:id="@+id/navigation_playlist"
             android:name="me.bickositieff.raspio.ui.playlist.PlaylistFragment"
             android:label="@string/title_playlist"
-            tools:layout="@layout/fragment_playlist"/>
+            tools:layout="@layout/fragment_playlist">
+        <action
+                android:id="@+id/add_song"
+                app:destination="@id/navigation_song_select"
+                app:popUpTo="@id/navigation_playlist"/>
```
Comment (profiluefter) [inline app/app/src/main/res/navigation/mobile_navigation.xml]: popUpTo sets where to go when you navigate back from the selection, so the previous screen (playlist) is already the default.
```suggestion
                app:destination="@id/navigation_song_select"/>
```

**Review by profiluefter — CHANGES_REQUESTED** [review]

**Review by oagna — APPROVED** [review]

**Review by profiluefter — APPROVED** [review]

**Pull request merged**